src/platform/premierp550.rs

```diff
@@ -0,0 +1,234 @@
+//! QEMU Virt board
+
+use core::fmt::Write;
+use core::{fmt, hint};
+
+use log::Level;
+use spin::Mutex;
+
+use crate::arch::{Arch, Architecture};
+use crate::config::{TARGET_FIRMWARE_ADDRESS, TARGET_START_ADDRESS};
+use crate::device::clint::{VirtClint, CLINT_SIZE};
+use crate::device::tester::{VirtTestDevice, TEST_DEVICE_SIZE};
+use crate::device::VirtDevice;
+use crate::driver::clint::ClintDriver;
+use crate::driver::uart::UartDriver;
+use crate::Platform;
+
+// —————————————————————————— Platform Parameters ——————————————————————————— //
+
+// TODO: Check about that if it is the case or not
+const MIRALIS_START_ADDR: usize = TARGET_START_ADDRESS;
+const FIRMWARE_START_ADDR: usize = TARGET_FIRMWARE_ADDRESS;
+
+// TODO: What is the base of the clint device?
```

Comment:
Just a note: this can be found in the SoC manual (so the Eswin [EIC7700X](https://www.sifive.com/document-file/eic7700x-datasheet))